docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,298 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. highlight:: console
+
+.. _building-arrow-java:
+
+===================
+Building Arrow Java
+===================
+
+.. contents::
+
+System setup
+============
+
+Arrow Java uses the `Maven <https://maven.apache.org/>`_ build system.
+
+Building requires:
+
+* JRE 8, 9, 10, 11
```

Comment:
JDK 8 or above?  Might be worth to mention versions above what is currently in tests aren't guaranteed to work.

Reply:
Changed. Mention: JRE 8,9,10,11 also mention that CI runs only for JRE11.